hazelcast-jet-core/src/main/java/com/hazelcast/jet/config/JobConfig.java

```diff
@@ -417,7 +570,14 @@ public JobConfig addResource(@Nonnull String path, @Nonnull String id) {
         return resourceConfigs;
     }
 
-    private JobConfig add(URL url, String id, ResourceType resourceType) {
+    private JobConfig add(@Nonnull URL url, @Nullable String id, @Nonnull ResourceType resourceType) {
```

Comment:
the `id` should be `@Nonnull` too, I checked all callers and they ensure it. Even if it's null, we pass it to `ResourceConfig` constructor where it's non-null.